### PR TITLE
INFINITY-2758/CASS-658 Document 'free' task env, and rack behavior

### PR DESCRIPTION
**Pull request by nickbp**
- There's already some region/zone docs, just needed some more for Cassandra.
- Add listing of freebie envvars that are automatically injected into tasks.

### CODE REVIEW

**Comment by nickbp** [comment]
ping

File: docs/pages/developer-guide.md [hunk 1]
```diff
@@ -1320,13 +1320,17 @@ REGION: us-west-2
...
 ```
 @zone:GROUP_BY:2
 ```
 
-The placement rule above would apply the `GROUP_BY` operator to zones. The SDK allows region-aware scheduling as a beta feature.  It may be enabled by setting the environment variable `ALLOW_REGION_AWARENESS` to `true`.  Once enabled, placement rules can be written that reference the `@region` key.
+The placement rule above would apply the `GROUP_BY` operator to zones.
+
+## Regions (beta)
+
+The SDK allows region-aware scheduling as a beta feature.  It may be enabled by setting the environment variable `ALLOW_REGION_AWARENESS` to `true`.  Once enabled, placement rules can be written that reference the `@region` key.
```
Comment (sascala) [inline docs/pages/developer-guide.md]: "Enable it by setting..."

File: docs/pages/developer-guide.md [hunk 2]
```diff
@@ -1946,14 +1950,45 @@ To be clear, the config templating provided by the `bootstrap` tool may be appli
 
 ### Task Environment
 
-While some environment variables are included by default in each task as a convenience, you may also specify custom environment variables yourself.  For example, a typical task would include the following key/value pairs in its environment:
+Task configuration is generally exposed using environment variables. A number environment variables describing the task and/or cluster environment are provided automatically, while others may be manually specified by the developer.
```
Comment (sascala) [inline docs/pages/developer-guide.md]: "A number of environment variables that describe the task and/or cluster environment are provided automatically, while the developer can manually specify others."

File: docs/pages/developer-guide.md [hunk 3]
```diff
@@ -1946,14 +1950,45 @@ To be clear, the config templating provided by the `bootstrap` tool may be appli
...
-POD_INSTANCE_INDEX: 0
-TASK_NAME: hello-0-server
-```
+#### Included Values
+
+The following environment values are automatically injected into all tasks as a convenience. These typically provide some additional context about the container.
+
+* `TASK_NAME=hello-3-server`
+
+The name of the task, such as `hello-3-server`
```
Comment (sascala) [inline docs/pages/developer-guide.md]: Period at the end of the sentence.

File: docs/pages/developer-guide.md [hunk 4]
```diff
@@ -1946,14 +1950,45 @@ To be clear, the config templating provided by the `bootstrap` tool may be appli
...
+
+The TLD for accessing tasks within the service. To address a given task, one could construct a hostname as `<taskname>.FRAMEWORK_HOST`. This varies according to the service name as configured by the user.
+
+* `FRAMEWORK_VIP_HOST=folderservicename.l4lb.thisdcos.directory`
+
+The TLD for VIPs advertised by the service. To address a given VIP, one could construct a hostname as `<vipname>.FRAMEWORK_VIP_HOST`. This varies according to the service name as configured by the user.
+
+* `SCHEDULER_API_HOSTNAME=api.folderservicename.marathon.l4lb.thisdcos.directory`
+
+The hostname where the Scheduler may be reached. Useful when tasks need to make API calls to a custom endpoint that's being run by the Scheduler.
```
Comment (sascala) [inline docs/pages/developer-guide.md]: "can be reached."

File: frameworks/cassandra/docs/node-settings.md [hunk 5]
```diff
@@ -90,6 +90,10 @@ Placement constraints allow you to customize where Apache Cassandra nodes are de
 *   **In DC/OS CLI options.json**: `placement_constraint`: string (default: `""`)
 *   **DC/OS web interface**: `PLACEMENT_CONSTRAINT`: `string`
 
+## Rack-Aware Placement
+
+Cassandra's "rack"-based fault domain support may be enabled by specifying a placement constraint which uses the `@zone` key. For example, one could spread Cassandra nodes across a minimum of three different zones/racks by specifying the constraint `@zone:GROUP_BY:3`. When a placement constraint specifying `@zone` is used, Cassandra nodes will be automatically configured with `rack`s which match the names of the zones. If no placement constraint referencing `@zone` is configured, all nodes will be configured with a default rack of `rack1`.
```
Comment (sascala) [inline frameworks/cassandra/docs/node-settings.md]: "constraint that uses..."

"with `racks that match..."

File: docs/pages/developer-guide.md (hunk 1 above)
Comment (nickbp) [reply docs/pages/developer-guide.md]: Done

File: docs/pages/developer-guide.md (hunk 2 above)
Comment (nickbp) [reply docs/pages/developer-guide.md]: Done

File: docs/pages/developer-guide.md (hunk 3 above)
Comment (nickbp) [reply docs/pages/developer-guide.md]: Done

File: docs/pages/developer-guide.md (hunk 4 above)
Comment (nickbp) [reply docs/pages/developer-guide.md]: Done

File: frameworks/cassandra/docs/node-settings.md (hunk 5 above)
Comment (nickbp) [reply frameworks/cassandra/docs/node-settings.md]: Done

**Pull request merged**